packages/miniflare/src/shared/dev-registry.ts

```diff
@@ -142,145 +260,60 @@ export class DevRegistry {
-				(durableObject) => durableObject.className === className
-			)
-		) {
-			return {
-				...target,
-				path: `/${INBOUND_DO_PROXY_SERVICE_PATH}`,
-			};
-		}
-
-		return null;
-	}
-
-	public subscribe(workerName: string, callback?: () => void): void {
-		let callbacks = this.subscribers.get(workerName);
-
-		if (!callbacks) {
-			callbacks = [];
-			this.subscribers.set(workerName, callbacks);
-		}
-
-		if (callback !== undefined) {
-			callbacks.push(callback);
-		}
-	}
-
-	private notifySubscribers(workerName: string): void {
-		const callbacks = this.subscribers.get(workerName);
-		if (callbacks) {
-			for (const callback of callbacks) {
-				callback();
-			}
-		}
-	}
-
 	private refresh(): void {
 		if (!this.registryPath) {
 			return;
 		}
 
-		// Make sure the registry path exists
-		mkdirSync(this.registryPath, { recursive: true });
-
-		const workerNames = readdirSync(this.registryPath);
+		this.registry = getWorkerRegistry(this.registryPath, (workerName) => {
+			this.unregister(workerName);
+		});
 
-		// Cleanup existing definition that are not in the registry anymore
-		for (const existingWorkerName of Object.keys(this.registry)) {
-			if (!workerNames.includes(existingWorkerName)) {
-				delete this.registry[existingWorkerName];
-				this.notifySubscribers(existingWorkerName);
-			}
+		// Send updated workers to the proxy worker
+		if (this.proxyWorker) {
+			this.proxyWorker.postMessage({
+				type: "update",
+				workers: this.registry,
+			});
 		}
+	}
+}
 
-		for (const workerName of workerNames) {
-			try {
-				const definitionPath = path.join(this.registryPath, workerName);
-				const file = readFileSync(definitionPath, "utf8");
-				const stats = statSync(definitionPath);
+export function getWorkerRegistry(
+	registryPath: string,
+	unregisterStaleWorker?: (workerName: string) => void
+): WorkerRegistry {
+	const registry: WorkerRegistry = {};
 
-				// Cleanup existing workers older than 5 minutes
-				if (stats.mtime.getTime() < Date.now() - 300_000) {
-					this.unregister(workerName);
-					continue;
-				}
+	if (!existsSync(registryPath)) {
+		return registry;
+	}
 
-				if (
-					// If the worker is not registered before, or
-					!this.registry[workerName] ||
-					// If the file content is different from the registry
-					file !== JSON.stringify(this.registry[workerName], null, 2)
-				) {
-					this.registry[workerName] = JSON.parse(file);
-					this.notifySubscribers(workerName);
-				}
-			} catch (e) {
-				// This can safely be ignored. It generally indicates the worker was too old and was removed by a parallel process
-				this.log?.debug(
-					`Error while loading worker definition from the registry: ${e}`
-				);
+	for (const workerName of readdirSync(registryPath)) {
+		try {
+			const definitionPath = path.join(registryPath, workerName);
+			const file = readFileSync(definitionPath, "utf8");
+			const stats = statSync(definitionPath);
+
+			// Cleanup existing workers older than 5 minutes
```

Comment:
I now this is how it works already but I am confused... does this mean that if you have a `wrangler dev` process running for more than 5 minutes it will get removed from the registry, even though it is still active?

Reply:
It's because we are updating the timestamp regularly when we register a worker through the dev registry: https://github.com/cloudflare/workers-sdk/blob/5d5ecd558d58461f203f882011df3e4d2652305c/packages/miniflare/src/shared/dev-registry.ts#L134-L141.

So this logic is meant to clean up stale workers in the registry that didn't get cleaned up properly.